src/Abp.RedisCache/Runtime/Caching/Redis/AbpRedisHashStore.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using JetBrains.Annotations;
+using StackExchange.Redis;
+
+namespace Abp.Runtime.Caching.Redis
+{
+    /// <summary>
+    /// store keys and values in redis as hash
+    /// </summary>
+    /// <typeparam name="TKey"></typeparam>
+    /// <typeparam name="TValue"></typeparam>
+    public class AbpRedisHashStore<TKey, TValue> : IAbpRedisHashStore<TKey, TValue>
+    {
+        private readonly Lazy<IDatabase> _database;
+        private readonly IRedisCacheSerializer _serializer;
+        public string StoreName { get; private set; }
+
+        public AbpRedisHashStore([NotNull]string storeName, IAbpRedisCacheDatabaseProvider databaseProvider, IRedisCacheSerializer serializer)
+        {
+            StoreName = "HashStore:" + storeName;
+            _serializer = serializer;
+            _database = new Lazy<IDatabase>(databaseProvider.GetDatabase);
+        }
+        public long Count => _database.Value.HashLength(StoreName);
+
+        public void Add(TKey key, TValue value)
+        {
+            if (ContainsKey(key))
+                throw new ArgumentException("An item with the same key has already been added");
```

Comment:
i believe always having curly brackets around if block is the convention

Reply:
😄😄 I believe writing less is the convention. But it depends on which is mostly used in the project. I will stick to the majority to make project look more orderly.

Reply:
We always use curly brackets around if blocks even if it is a single line.

Reply:
Ok, i will stick to the majority. 👍